docs/golang_extension.md

```diff
@@ -0,0 +1,93 @@
+# Golang (plugin) extensions
+
+Gohan supports extensions written in golang and imported as shared libraries.
+Golang plugins are supported starting from version 1.8 of golang. Note that
+there exist a bug [1] which will be fixed in version 1.9.
+
+## Why golang extensions?
+
+One of the main reasons to choose a golang extensions over a javascript ones is the
+performance. Golang plugins are compiled to native code (not interpreted as javascript
+extensions) and run fast. In contrast to javascript, golang is a typed language which
+turns out to be very helpful for developers. It is also possible to use a debugger
+to find problems in code easily.
+
+# Creating a simple golang extension
+
+## Extension structure
+
+An extension consists of an init.go file placed in loader subdirectory
+and a number of implementation files.
+
+# Schema definitions
+
+To enable a golang extension, append extensions section to schema and specify
+**code_type** as **goext**. Specify plugin library in url entry. Note that placing
+golang code in code entry is not supported.
+
+```yaml
+  extensions:
+  - id: example_golang_extension
+    code_type: goext
+    path: ""
+    url: file://golang_ext.so
+```
+
+## Required exports
+
+Each plugin must export Init function.
+
+```go
+func Init(env goext.IEnvironment) error {
+    schema := env.Schemas().Find("resource_schema")
+    schema.RegisterRawType(resource.Resource{})
```

Comment:
Maybe we should include simple but complete init.go file including import statements. We can simply copy example.go .